src/Xrefcheck/Verify.hs

```diff
@@ -426,94 +454,103 @@ verifyReference
                                   </> toString (canonizeLocalRef rLink))
           AbsoluteLoc       -> checkRef rAnchor (root <> toString rLink)
           ExternalLoc       -> checkExternalResource config rLink
-          OtherLoc          -> verifying pass
-        else return mempty
+          OtherLoc          -> pure Nothing
+        else pure Nothing
   where
-    retryVerification
-      :: Int
-      -> IO (VerifyResult VerifyError)
-      -> IO (VerifyResult $ WithReferenceLoc VerifyError)
-    retryVerification numberOfRetries resIO = do
-      res@(VerifyResult ves) <- resIO
+    addReference :: VerifyError -> WithReferenceLoc VerifyError
+    addReference = WithReferenceLoc fileWithReference ref
 
+    retryVerification
+      :: RetryCounter
+      -> IO (Maybe VerifyError)
+      -> IO (Maybe VerifyError)
```

Comment:
In fact we get the maximum of one error from the checking functions, so using here newtype above the list was a bit confusing IMO. Also I refactored this function in order to isolate different cases from each other and make code more readable.

Reply:
I like the resulting code after the refactor! It seems easier to understand. However, let's wait for @Martoon-00 's review because I don't know som much this code and maybe he can catch anything that could have been unintentionally changed.

Reply:
> But I'm relatively skeptical about switching to `Maybe`, because in my understanding, the commonly taken semantics of `Maybe` is: `Just` for ok and `Nothing` for failure. To carry errors, `Either err ()` or some custom type suits more thinking
I think that we can slightly improve the code if we rename `res` to `err`, so we'll show that we expect an error, and `Nothing` will just denote that there was no errors. As for me, there is not really big difference between `Maybe err` and `Either err ()` because here we don't expect anything except unit.

Reply:
Sure, I'll then replace `Maybe e` with `Either e ()`

Reply:
Thanks :+1: